pkg/network/netbinding/memory.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package netbinding
+
+import (
+	k8scorev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+func CalculateMemoryOverhead(vmi *v1.VirtualMachineInstance, registeredPlugins map[string]v1.InterfaceBindingPlugin) resource.Quantity {
+	return sumPluginsMemoryOverhead(
+		filterUniquePlugins(vmi.Spec.Domain.Devices.Interfaces, registeredPlugins),
+	)
+}
+
+func filterUniquePlugins(interfaces []v1.Interface, registeredPlugins map[string]v1.InterfaceBindingPlugin) []v1.InterfaceBindingPlugin {
```

Comment:
I think you can simplify it, please consider by changing this function to return the unique-plugin map, and  `sumPluginsMemoryOverhead` to receive a map instead.

This way there is no need for both unique map and a slice, it could also make it easier to follow IMO.

Reply:
> I think you can simplify it, please consider by changing this function to return the unique-plugin map, and `sumPluginsMemoryOverhead` to receive a map instead.
> 
> This way there is no need for both unique map and a slice, it could also make it easier to follow IMO.

This was implemented in one of the early iterations.
I received a suggestion from @EdDev to change it to the current implementation.

In any case we could change it in the follow-up that will extract this code from `template.go`.